src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -34,21 +40,47 @@ public EventPipeEventSource(string fileName)
             cpuSpeedMHz = 10;
 
             _deserializer = new Deserializer(new PinnedStreamReader(fileName, 0x20000), fileName);
+
+#if SUPPORT_V1_V2
+            // This is only here for V2 and V1.  V3+ should use the name EventTrace, it can be removed when we drop support.
             _deserializer.RegisterFactory("Microsoft.DotNet.Runtime.EventPipeFile", delegate { return this; });
-        
-            var entryObj = _deserializer.GetEntryObject(); // this call invokes FromStream and reads header data
+#endif
+            _deserializer.RegisterFactory("!EventTrace.", delegate { return this; });
+            _deserializer.RegisterFactory("!EventBlock.", delegate { return new EventPipeEventBlock(this); });
 
-            // Because we told the deserialize to use 'this' when creating a EventPipeFile, we 
-            // expect the entry object to be 'this'.
-            Debug.Assert(entryObj == this);
```

Comment:
Why did we remove the assert?  (of course it goes after 'entryObj' is defined, but it is still correct and valuable (as well as the comment.

Reply:
I removed this assert because I implemented unit tests to verify the correct behavior of this code. ([CanParseHeaderOfV1EventPipeFile](https://github.com/Microsoft/perfview/blob/fd07903405f3c7f88b8030711575ec1a5a5d6473/src/TraceEvent/TraceEvent.Tests/EventPipeParsing.cs#L69) and [CanParseHeaderOfV3EventPipeFile](https://github.com/Microsoft/perfview/blob/fd07903405f3c7f88b8030711575ec1a5a5d6473/src/TraceEvent/TraceEvent.Tests/EventPipeParsing.cs#L92))

imho we should have more unit tests and fewer debug.asserts (I understand that we use Debug.Assert here as a debug-only code contracts) to make the code more easy to understand and maintain.

I reverted this change.